protobuf/lib/src/protobuf/builder_info.dart

```diff
@@ -323,13 +318,12 @@ class BuilderInfo {
 
   ProtobufEnum? _decodeEnum(
       int tagNumber, ExtensionRegistry? registry, int rawValue) {
-    final f = valueOfFunc(tagNumber);
-    if (f != null) {
-      return f(rawValue);
+    final valueMap = fieldInfo[tagNumber]?.enumValueMap;
+    if (valueMap != null) {
+      return valueMap[rawValue];
```

Comment:
The benchmarks show this PR is slower than the other. This isn't surprising as we are using a map index here and a list in the other PR.

Now what prevents this PR from using a list instead of a map?

Right now this PR passes in the map in the `BuilderInfo` methods, but it may as well pass a list, why not?

But as mentioned in the other comment we may not need to pass any redundant information in at all.

Reply:
Answered in the other comment.